docs/tutorials/11_GMPHDTutorial.py

```diff
@@ -0,0 +1,577 @@
+        z += weights[gaussian]*multivariate_normal.pdf(x=quantiles, mean=means[gaussian,:], cov=sigmas[gaussian])
+    return z
+
+
+# %%
+# For each timestep, create a new figure with 2 subplots. The plot on the left will
+# show the 3D Gaussian Mixture density. The plot on the right will show a 2D
+# birds-eye-view of the space, including the ground truths, detections, clutter, and tracks.
+
+from matplotlib import animation
+from matplotlib import pyplot as plt
+from matplotlib.lines import Line2D # Will be used when making the legend
+
+# This is the function that updates the figure we will be animating. As parametrs we must
+# pass in the elements that will be changed, as well as the index i
+def animate(i, sf, truths, tracks, measurements, clutter):
+    # Set up the axes
+    axL.clear()
+    axR.set_title('Tracking Space at k='+str(i))
+    axL.set_xlabel("x")
+    axL.set_ylabel("y")
+    axL.set_title('PDF of the Gaussian Mixture')
+    axL.view_init(elev=30, azim=-80)
+    axL.set_zlim(0, 0.3)
+
+    # Initialize the variables
+    num_gaussians = len(all_gaussians[i]) # number of functions in this mixture
+    weights = []  # weights of each Gaussian. This is analogous to the probability of its existence
+    means = []    # means of each Gaussian. This is equal to the x and y of its state vector
+    sigmas = []   # standard deviation of each Gaussian.
+
+    # Fill the lists of weights, means, and standard deviations
+    for state in all_gaussians[i]:
+        weights.append(state.weight)
+        means.append([state.state_vector[0], state.state_vector[2]])
+        sigmas.append([state.covar[0][0], state.covar[1][1]])
+    means = np.array(means)
+    sigmas = np.array(sigmas)
+
+    # Generate the z values over the space and plot on the left axis
+    zarray[:,:,i] = get_mixture_density(x, y, weights, means, sigmas)
+    sf = axL.plot_surface(x, y, zarray[:,:,i], cmap=cm.RdBu, linewidth=0, antialiased=False)
+
+    # Make lists to hold the new ground truths, tracks, detections, and clutter
+    new_truths, new_tracks, new_measurements, new_clutter = [], [], [], []
+    for truth in truths_by_time[i]:
+        new_truths.append([truth.state_vector[0], truth.state_vector[2]])
+    for state in tracks_by_time[i]:
+        new_tracks.append([state.state_vector[0], state.state_vector[2]])
+    for measurement in all_measurements[i]:
+        if type(measurement) == TrueDetection:
+            new_measurements.append([measurement.state_vector[0], measurement.state_vector[1]])
+        elif type(measurement == Clutter):
+            new_clutter.append([measurement.state_vector[0], measurement.state_vector[1]])
+
+    # Plot the contents of these lists on the right axis
+    if len(new_truths) > 0: truths.set_offsets(new_truths)
+    if len(new_tracks) > 0: tracks.set_offsets(new_tracks)
+    if len(new_measurements) > 0: measurements.set_offsets(new_measurements)
+    if len(new_clutter) > 0: clutter.set_offsets(new_clutter)
```

Comment:
```suggestion
    if new_truths:
        truths.set_offsets(new_truths)
    if new_tracks:
        tracks.set_offsets(new_tracks)
    if new_measurements:
        measurements.set_offsets(new_measurements)
    if new_clutter:
        clutter.set_offsets(new_clutter)
```
Best practice for checking if a sequence is empty.